src/PowerShellEditorServices.Protocol/MessageProtocol/Channel/ChannelBase.cs

```diff
@@ -0,0 +1,118 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.Serializers;
+using Newtonsoft.Json.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.Channel
+{
+    /// <summary>
+    /// Defines a base implementation for servers and their clients over a
+    /// single kind of communication channel.
+    /// </summary>
+    public abstract class ChannelBase
```

Comment:
This is great. I love how simple it is. I think it will make it much easier to implement additional channels.